code/game/objects/effects/decals/cleanable/misc.dm

```diff
@@ -509,6 +514,32 @@
 		ignite()
 	return ..()
 
+/obj/effect/decal/cleanable/fuel_pool/on_entered(datum/source, atom/movable/AM)
+	. = ..()
+	if(AM.throwing) // don't light from things being thrown over us
+		return
+	if(isitem(AM))
+		var/obj/item/item = AM
+		if(item.get_temperature() > FIRE_MINIMUM_TEMPERATURE_TO_EXIST)
+			ignite()
+	else if(isliving(AM))
+		var/mob/living/L = AM
+		if(L.on_fire)
+			ignite()
+
+/obj/effect/decal/cleanable/fuel_pool/proc/ignition_trigger(datum/source, atom/movable/AM)
+	SIGNAL_HANDLER
+
+	if(isitem(AM))
+		var/obj/item/item = AM
+		if(item.get_temperature() > FIRE_MINIMUM_TEMPERATURE_TO_EXIST)
+			ignite()
+	else if(isliving(AM))
+		var/mob/living/L = AM
+		if(L.on_fire)
+			ignite()
```

Comment:
Change these single letter variables to more sensible ones.
Also: This code is the same as the preceding block. Have that call this proc, or alternately add a third proc which both of these procs call.